system/src/Grav/Common/Grav.php

```diff
@@ -215,83 +115,13 @@ public function process()
         /** @var Debugger $debugger */
         $debugger = $this['debugger'];
 
-        // Load site setup and initializing streams.
-        $debugger->startTimer('_setup', 'Site Setup');
-        $this['setup']->init();
-        $this['streams'];
-        $debugger->stopTimer('_setup');
-
-        // Initialize configuration.
-        $debugger->startTimer('_config', 'Configuration');
-        /** @var Plugins $plugins */
-        $plugins = $this['plugins']->setup();
-        $this['config']->init();
-        $debugger->stopTimer('_config');
-
-        // Initialize error handlers.
-        $this['errors']->resetHandlers();
-
-        // Initialize debugger.
-        $debugger->init();
-        $debugger->startTimer('init', 'Initialize');
-        $this['config']->debug();
-
-        // Use output buffering to prevent headers from being sent too early.
-        ob_start();
-        if ($this['config']->get('system.cache.gzip')) {
-            // Enable zip/deflate with a fallback in case of if browser does not support compressing.
-            if (!ob_start("ob_gzhandler")) {
-                ob_start();
-            }
-        }
-
-        // Initialize the timezone.
-        if ($this['config']->get('system.timezone')) {
-            date_default_timezone_set($this['config']->get('system.timezone'));
-        }
-
-        // Initialize uri, session.
-        $this['uri']->init();
-        $this['session']->init();
-
-        $this->setLocale();
-
-        $debugger->stopTimer('init');
-
-        $debugger->startTimer('plugins', 'Plugins');
-        $plugins->init();
-        $this->fireEvent('onPluginsInitialized');
-        $debugger->stopTimer('plugins');
-
-        $debugger->startTimer('themes', 'Themes');
-        $this['themes']->init();
-        $debugger->stopTimer('themes');
-
-        $task = $this['task'];
-        if ($task) {
-            $this->fireEvent('onTask.' . $task);
-        }
-
-        $this['assets']->init();
-        $this->fireEvent('onAssetsInitialized');
-
-        $debugger->startTimer('twig', 'Twig');
-        $this['twig']->init();
-        $debugger->stopTimer('twig');
-
-        $debugger->startTimer('pages', 'Pages');
-        $this['pages']->init();
-        $this->fireEvent('onPagesInitialized');
-        $debugger->stopTimer('pages');
-        $this->fireEvent('onPageInitialized');
-
-        $debugger->addAssets();
-
-        // Process whole page as required
-        $debugger->startTimer('render', 'Render');
-        $this->output = $this['output'];
-        $this->fireEvent('onOutputGenerated');
-        $debugger->stopTimer('render');
+        // process all processors (e.g. config, initialize, assets, ..., render)
+        foreach ($this->processors as $processor):
+          $processor = $this[$processor];
+          $this->measureTime($processor->id, $processor->title, function($debugger) use ($processor) {
+            $processor->process($debugger);
+          });
+        endforeach;
```

Comment:
As the other code uses the classical notation { ... } I have changed it. But I've not see any guideline that say the alternative syntax should only be used in templates. Happy to learn something. :)